# Data Cleaning

## Objective

The objective of this notebook is to prepare the integrated adaptations dataset for exploratory data analysis by removing invalid records, correcting inconsistencies, and standardizing the dataset.

## Cleaning Steps

* Removed records marked for exclusion during manual verification.
* Replaced incorrect book titles with manually corrected values.
* Removed unnecessary columns used during the data collection process.
* Converted release dates to datetime format.
* Verified missing values and duplicate records.
* Created the final cleaned dataset for analysis.


In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('../data/books/final/updated_adaptations.csv')

df.info()

<class 'pandas.DataFrame'>
RangeIndex: 268 entries, 0 to 267
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Title               268 non-null    str    
 1   Premiere            268 non-null    str    
 2   Rated               252 non-null    str    
 3   imdb_rating         268 non-null    float64
 4   imdb_votes          268 non-null    int64  
 5   imdb_genre          268 non-null    str    
 6   imdb_year           268 non-null    int64  
 7   imdb_type           268 non-null    str    
 8   is_book_adaptation  268 non-null    str    
 9   book_title          268 non-null    str    
 10  book_author         268 non-null    str    
 11  search_key          268 non-null    str    
 12  platform            268 non-null    str    
 13  goodreads_rating    249 non-null    float64
 14  keep                268 non-null    str    
dtypes: float64(2), int64(2), str(11)
memory usage: 31.5 KB


In [3]:
df.head()

,Title,Premiere,Rated,imdb_rating,imdb_votes,imdb_genre,imdb_year,imdb_type,is_book_adaptation,book_title,book_author,search_key,platform,goodreads_rating,keep
0,Reacher,04/02/2022,TV-MA,8.0,291841,"Action, Crime, Drama",2022,series,y,jack reacher series,lee child,jack reacher,Amazon,4.08,y
1,The Summer I Turned Pretty,17/06/2022,TV-14,7.2,74872,"Drama, Romance",2022,series,y,the summer i turned pretty,jenny han,the summer i turned pretty,Amazon,3.73,y
2,The Terminal List,01/07/2022,TV-MA,7.9,127044,"Action, Drama, Thriller",2022,series,y,the terminal list,jack carr,the terminal list,Amazon,4.29,y
3,The Lord of the Rings: The Rings of Power,01/09/2022,TV-14,6.9,436113,"Action, Adventure, Drama",2022,series,y,the lord of the rings and related middle-earth...,j. r. r. tolkien,the lord of the rings and related middle-earth...,Amazon,4.50,y
4,Cross,14/11/2024,TV-MA,7.2,34967,"Action, Crime, Drama",2024,series,y,alex cross series,james patterson,alex cross,Amazon,4.13,y


## Removing Non-Adaptation Titles

In [4]:
df.value_counts('keep')

keep
y    249
n     19
Name: count, dtype: int64

In [5]:
df = df[df['keep'].str.lower() == 'y']

## Correcting Book Titles

In [6]:
df = df.drop(columns=['book_title'])

In [7]:
df = df.rename(columns={'search_key': 'book_title'})

## Dropping Unnecessary Columns

In [8]:
df = df.drop(columns=['is_book_adaptation','keep'])

In [9]:
df = df.reset_index(drop=True)

In [10]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 249 entries, 0 to 248
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Title             249 non-null    str    
 1   Premiere          249 non-null    str    
 2   Rated             236 non-null    str    
 3   imdb_rating       249 non-null    float64
 4   imdb_votes        249 non-null    int64  
 5   imdb_genre        249 non-null    str    
 6   imdb_year         249 non-null    int64  
 7   imdb_type         249 non-null    str    
 8   book_author       249 non-null    str    
 9   book_title        249 non-null    str    
 10  platform          249 non-null    str    
 11  goodreads_rating  249 non-null    float64
dtypes: float64(2), int64(2), str(8)
memory usage: 23.5 KB


## Converting Data Types

In [11]:
df['Premiere'] = pd.to_datetime(df['Premiere'], format='%d/%m/%Y')

## Final Dataset Validation

In [12]:
df = df [
    [
        "Title",
        "Premiere",
        "Rated",
        "imdb_rating",
        "imdb_votes",
        "imdb_genre",
        "imdb_year",
        "imdb_type",
        "platform",
        "book_title",
        "book_author",
        "goodreads_rating"
    ]
]

In [13]:
df = df.reset_index(drop=True)

In [14]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 249 entries, 0 to 248
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   Title             249 non-null    str           
 1   Premiere          249 non-null    datetime64[us]
 2   Rated             236 non-null    str           
 3   imdb_rating       249 non-null    float64       
 4   imdb_votes        249 non-null    int64         
 5   imdb_genre        249 non-null    str           
 6   imdb_year         249 non-null    int64         
 7   imdb_type         249 non-null    str           
 8   platform          249 non-null    str           
 9   book_title        249 non-null    str           
 10  book_author       249 non-null    str           
 11  goodreads_rating  249 non-null    float64       
dtypes: datetime64[us](1), float64(2), int64(2), str(7)
memory usage: 23.5 KB


In [15]:
df.head()

,Title,Premiere,Rated,imdb_rating,imdb_votes,imdb_genre,imdb_year,imdb_type,platform,book_title,book_author,goodreads_rating
0,Reacher,2022-02-04,TV-MA,8.0,291841,"Action, Crime, Drama",2022,series,Amazon,jack reacher,lee child,4.08
1,The Summer I Turned Pretty,2022-06-17,TV-14,7.2,74872,"Drama, Romance",2022,series,Amazon,the summer i turned pretty,jenny han,3.73
2,The Terminal List,2022-07-01,TV-MA,7.9,127044,"Action, Drama, Thriller",2022,series,Amazon,the terminal list,jack carr,4.29
3,The Lord of the Rings: The Rings of Power,2022-09-01,TV-14,6.9,436113,"Action, Adventure, Drama",2022,series,Amazon,the lord of the rings and related middle-earth...,j. r. r. tolkien,4.50
4,Cross,2024-11-14,TV-MA,7.2,34967,"Action, Crime, Drama",2024,series,Amazon,alex cross,james patterson,4.13


In [16]:
df.to_csv('../data/final/book_adaptations.csv', index=False)

## Final Dataset Summary

The final cleaned dataset contains 248 verified book adaptations from Netflix and Amazon Prime Video, enriched with IMDb and Goodreads ratings and prepared for exploratory data analysis.